src/functions.jl

```diff
@@ -64,10 +53,18 @@ function invdivrem((q, r), divisor)
     end
 end
 
-function invfldmod((q, r), divisor)
+function invfldmod((q, r)::NTuple{2,Number}, divisor::Number)
     if abs(r) ≤ abs(divisor) && (iszero(r) || sign(r) == sign(divisor))
         muladd(q, divisor, r)
     else
         throw(DomainError((q, r), "inverse for fldmod(x) is not defined at this point"))
     end
 end
+
+
+# check if T is a real-Number type
+# this is not the same as T <: Real which immediately excludes custom Number subtypes such as unitful numbers
+# also, isreal(x) != is_real_type(typeof(x)): the former is true for complex numbers with zero imaginary part
+is_real_type(@nospecialize _::Type{<:Real}) = true
+is_real_type(::Type{T}) where {T<:Number} = real(T) == T
+is_real_type(_) = false
```

Comment:
```suggestion
# check if T is a real-Number type
# this is not the same as T <: Real which immediately excludes custom Number subtypes such as unitful numbers
# also, isreal(x) != is_real_type(typeof(x)): the former is true for complex numbers with zero imaginary part
is_real_type(@nospecialize _::Type{<:Real}) = Val(true)
is_real_type(::Type{T}) where {T<:Number} = Val(real(T) == T)
is_real_type(@nospecialize _::Any) = Val(false)
```

Reply:
Yes, I think it would be better, constant propagation doesn't always work.

Reply:
Well, if we make the functions @inline, maybe that'll be enough. Ok, let's try returning values instead of `Val`s.